firstvoices/backend/tests/test_search/test_filters.py

```diff
@@ -207,3 +207,35 @@ def test_staff_permissions(self, role):
         search_query = search_query.to_dict()
 
         assert self.public_permissions_filter not in str(search_query)
+
+
+class TestVisibilityParam:
+    def test_default(self):
+        search_query = get_search_query()
+        search_query = search_query.to_dict()
+
+        assert "filter" not in search_query["query"]["bool"] or "visibility" not in str(
+            search_query["query"]["bool"]["filter"]
+        )
+
+    @pytest.mark.parametrize(
+        "visibility",
+        [
+            [Visibility.PUBLIC],
+            [Visibility.MEMBERS],
+            [Visibility.TEAM],
+            [Visibility.PUBLIC, Visibility.MEMBERS],
+            [Visibility.PUBLIC, Visibility.TEAM],
+            [Visibility.MEMBERS, Visibility.TEAM],
+            [Visibility.PUBLIC, Visibility.MEMBERS, Visibility.TEAM],
+        ],
+    )
+    def test_team(self, visibility):
+        search_query = get_search_query(visibility=visibility)
+        search_query = search_query.to_dict()
+
+        filtered_terms = search_query["query"]["bool"]["filter"][0]["terms"]
+        assert "visibility" in filtered_terms
+
+        for value in visibility:
+            assert value in filtered_terms["visibility"]
```

Comment:
Please also assert that visibility and filtered_terms["visibility"] have the same length. This test would currently pass if the query always included all visibility options.